### ISSUE TITLE
The syslog logging target uses an empty certificate and should not be verified successfully.

### ISSUE THREAD

**Issue by kingsd041** [rancher/rancher#18470]
**What kind of request is this (question/bug/enhancement/feature request):**bug

**Rancher version:** rancher/rancher:master 02/27

**Prerequisites:**
Rsyslog server has enabled certificate verification
```
module(load="imtcp" StreamDriver.Name="gtls" StreamDriver.Mode="1" StreamDriver.AuthMode="x509/certvalid")
```

**Steps to reproduce (least amount of steps as possible):**
1. Configure the syslog logging target
2. Endpoint Select `TCP` and enter the correct syslog ip and port
3. Enable `Use TLS`
4. Click `TEST` when `Client Private Key` and `Client Certificate` are empty.
**Result:**
Returns `Settings validated` on the UI. This is wrong because the syslog server has returned a log of certificate validation failures.
```
rsyslogd: gnutls returned error on handshake: An unexpected TLS packet was received.  [v8.34.0 try http://www.rsyslog.com/e/2083 ]
2019-02-28T06:59:26.343181+00:00 7ee7c56258ff rsyslogd: gnutls returned error on handshake: An unexpected TLS packet was received.  [v8.34.0 try http://www.rsyslog.com/e/2083 ]
```

**Comment by aiwantaozi** [comment]
Fixed in 2.2

**Comment by kingsd041** [comment]
Tested with rancher/rancher version- v2.2.0 
Using the empty certificate `TEST`, the UI will prompt 
```
couldn't read data from syslog michelia.cnrancher.com:60515: couldn't read data from remote server: read tcp 10.42.0.6:59894->47.75.9.20:60515: Read: connection reset by peer
```
